app/lib/dartdoc/backend.dart

```diff
@@ -86,6 +86,23 @@ class DartdocBackend {
     return new PubDartdocData.fromJson(map);
   }
 
+  /// Deletes the old entries that predate [shared_versions.gcBeforeRuntimeVersion].
+  Future deleteOldSdkData() async {
+    final prefix = storage_path.dartSdkDartdocPrefix();
+    await for (BucketEntry entry in _storage.list(prefix: '$prefix/')) {
+      if (entry.isDirectory) {
+        continue;
+      }
+      final name = p.basename(entry.name);
+      final version = name.replaceAll('.json.gz', '');
+      if (version.length == 10 &&
+          version.compareTo('2018.00.00') > 0 &&
```

Comment:
Why this check? What about older data?

Reply:
Sanity/paranoid check, so that we don't have a similarly-looking file that we delete by accident.

Reply:
How about checking if it matches the regular expression for a date?